circuit/src/ecc/generic/native.rs

```diff
@@ -274,6 +274,81 @@ where
 	}
 }
 
+/// Structure for the UnassignedEcPoint
+#[derive(Clone, Debug)]
+pub struct UnassignedEcPoint<
+	C: CurveAffine,
+	N: FieldExt,
+	const NUM_LIMBS: usize,
+	const NUM_BITS: usize,
+	P,
+	EC,
+> where
+	P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS> + RnsParams<C::ScalarExt, N, NUM_LIMBS, NUM_BITS>,
+	EC: EccParams<C>,
+	<C as CurveAffine>::Base: FieldExt,
+	<C as CurveAffine>::ScalarExt: FieldExt,
+{
+	/// X coordinate of the UnassignedEcPoint
+	pub x: UnassignedInteger<C::Base, N, NUM_LIMBS, NUM_BITS, P>,
+	/// Y coordinate of the UnassignedEcPoint
+	pub y: UnassignedInteger<C::Base, N, NUM_LIMBS, NUM_BITS, P>,
+
+	_ec: PhantomData<EC>,
+}
+
+impl<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P, EC>
+	UnassignedEcPoint<C, N, NUM_LIMBS, NUM_BITS, P, EC>
+where
+	P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS> + RnsParams<C::ScalarExt, N, NUM_LIMBS, NUM_BITS>,
+	EC: EccParams<C>,
+	<C as CurveAffine>::Base: FieldExt,
+	<C as CurveAffine>::ScalarExt: FieldExt,
+{
+	/// Creates a new unassigned ec point object
+	pub fn new(
+		x: UnassignedInteger<C::Base, N, NUM_LIMBS, NUM_BITS, P>,
+		y: UnassignedInteger<C::Base, N, NUM_LIMBS, NUM_BITS, P>,
+	) -> Self {
+		Self { x, y, _ec: PhantomData }
+	}
+}
+
+impl<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P, EC>
+	From<EcPoint<C, N, NUM_LIMBS, NUM_BITS, P, EC>>
+	for UnassignedEcPoint<C, N, NUM_LIMBS, NUM_BITS, P, EC>
+where
+	P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS> + RnsParams<C::ScalarExt, N, NUM_LIMBS, NUM_BITS>,
+	EC: EccParams<C>,
+	<C as CurveAffine>::Base: FieldExt,
+	<C as CurveAffine>::ScalarExt: FieldExt,
+{
+	fn from(ec_point: EcPoint<C, N, NUM_LIMBS, NUM_BITS, P, EC>) -> Self {
+		Self {
+			x: UnassignedInteger::from(ec_point.x),
+			y: UnassignedInteger::from(ec_point.y),
+			_ec: PhantomData,
+		}
+	}
+}
+
+impl<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P, EC>
+	UnassignedValue for UnassignedEcPoint<C, N, NUM_LIMBS, NUM_BITS, P, EC>
+where
+	P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS> + RnsParams<C::ScalarExt, N, NUM_LIMBS, NUM_BITS>,
+	EC: EccParams<C>,
+	<C as CurveAffine>::Base: FieldExt,
+	<C as CurveAffine>::ScalarExt: FieldExt,
+{
+	fn without_witnesses() -> Self {
+		Self {
+			_ec: PhantomData,
+			x: UnassignedInteger::without_witnesses(),
+			y: UnassignedInteger::without_witnesses(),
+		}
+	}
+}
+
```

Comment:
This should be in mod.rs

Reply:
I mean generic/mod.rs

Reply:
I moved all the `Unassigned...` types to `mod.rs` files